### ISSUE TITLE
Fix TypeError error "Cannot destructure property..." in custom-property-pattern

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#5972]
### What steps are needed to reproduce the bug?

```css
a { prop: var(); }
```

[Demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYKAOAnCH4oDcksAKASgG4UBfAHQDsQAaEAZwE96AXJADwXCtWzcBHoAzAJYBzAcAYoUtEFgCuAGxitl+efUWLlYVay4QAtgFpsuGFi7trSLlzv0dKANoKDhkAD0STyRLAC8AXWCwgAZLAE5wgCoyEkso0NiEgGoyRIASZSYfXz1fX2VzLVYkaRgPZQBRXgwYMFcoFGNTC0wcFvt2FHokSpQzFAAjGBQAaxgJpAnLMCRWOpBigzp9A3CfbeoQaiA)

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "custom-property-pattern": [
      "^([a-z][a-z0-9]*)(-[a-z0-9]+)*$",
      {
        "message": "Expected custom property name to be kebab-case"
      }
    ]
  }
}
```

### How did you run Stylelint?

Demo

### Which version of Stylelint are you using?

14.6.0

### What did you expect to happen?

No TypeError

### What actually happened?

TypeError

### Does the bug relate to non-standard syntax?

No

### Proposal to fix the bug

Check that `nodes` is available before destructuring.

Follow-up PR to add "empty functions" to [commonly overlooked edge cases](https://stylelint.io/developer-guide/rules#commonly-overlooked-edge-cases).

**Comment by mattxwang** [comment]
Submitted a quick fix, though I check if `nodes[0]` exists rather than `nodes` (checking for `nodes` doesn't resolve the problem). Let me know what you think!
